Samples.Objc/Samples.ObjC/ViewControllers/CheckoutViewController/CheckoutViewController.m

```diff
@@ -0,0 +1,353 @@
+                                                                    items:self.items
+                                                                 shipping:nil];
+  NSArray<PPCPurchaseUnit *> *purchaseUnits = @[purchaseUnit];
+  PPCOrderRequest *order = [[PPCOrderRequest alloc] initWithIntent:PPCOrderIntentAuthorize
+                                                     purchaseUnits:purchaseUnits
+                                                             payer:nil
+                                                applicationContext:nil];
+  return order;
+}
+
+- (void)setupTableView {
+  [self.tableView registerClass:[AddItemCell class] forCellReuseIdentifier:@"AddItemCell"];
+  self.tableView.delegate = self;
+  self.tableView.dataSource = self;
+}
+
+- (void)setupSampleItem {
+  PPCPurchaseUnitItem *item = [[PPCPurchaseUnitItem alloc] initWithName:@"Sample item"
+                                                             unitAmount:[[PPCUnitAmount alloc] initWithCurrencyCode:PPCCurrencyCodeUsd value:@"5.99"]
+                                                               quantity:@"1"
+                                                                    tax:nil
+                                                        itemDescription:nil
+                                                                    sku:nil
+                                                               category:PPCPurchaseUnitCategoryPhysicalGoods];
+  self.items = [[NSMutableArray alloc] init];
+  [self.items addObject:item];
+}
+
+- (void)setupUI {
+  self.view.backgroundColor = [UIColor whiteColor];
+  
+  NSArray *options = @[@"Order", @"ECToken"];
+  self.segmentedControl = [[UISegmentedControl alloc] initWithItems:options];
+  [self.segmentedControl addTarget:self action:@selector(segmentChanged:) forControlEvents:UIControlEventValueChanged];
+  self.segmentedControl.selectedSegmentIndex = 0;
+  self.segmentedControl.translatesAutoresizingMaskIntoConstraints = false;
+  
+  self.tableView = [[UITableView alloc] init];
+  self.tableView.translatesAutoresizingMaskIntoConstraints = false;
+  
+  self.checkoutButton = [[UIButton alloc] init];
+  self.checkoutButton.layer.cornerRadius = 8;
+  self.checkoutButton.backgroundColor = [UIColor systemBlueColor];
+  self.checkoutButton.tintColor = [UIColor whiteColor];
+  [self.checkoutButton setTitle:@"Checkout with order" forState:UIControlStateNormal];
+  [self.checkoutButton.titleLabel setFont:[UIFont boldSystemFontOfSize:16]];
+  [self.checkoutButton addTarget:self action:@selector(tapCheckout) forControlEvents:UIControlEventTouchUpInside];
+  self.checkoutButton.translatesAutoresizingMaskIntoConstraints = false;
+    
+  [self.view addSubview:self.segmentedControl];
+  [self.view addSubview:self.tableView];
+  [self.view addSubview: self.checkoutButton];
+  
+}
+
+- (void)setupConstraints {
+  [[self.segmentedControl.topAnchor constraintEqualToAnchor:self.view.topAnchor constant:40] setActive:true];
```

Comment:
Could we use something like `view.safeAreaLayoutGuide.topAnchor` instead for this? Right now, running on a iPhone 12, this is right under the notch. Using the safeAreaLayoutGuide should allow us consistent padding across each devices safe area.